deeplake/core/chunk/chunk_compressed_chunk.py

```diff
@@ -35,16 +36,198 @@ def __init__(self, *args, **kwargs):
         self._changed = False
         self._compression_ratio = 0.5
 
-    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True) -> float:  # type: ignore
+    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True, lengths: Optional[List[int]] = None) -> float:  # type: ignore
         self.prepare_for_write()
+        if lengths is not None:  # this is triggered only for htype == "text"
+            return self.extend_if_has_space_byte_compression_text(
+                incoming_samples, update_tensor_meta, lengths
+            )
         if self.is_byte_compression:
+            if isinstance(incoming_samples, np.ndarray):
+                return self.extend_if_has_space_byte_compression_numpy(
+                    incoming_samples, update_tensor_meta
+                )
             return self.extend_if_has_space_byte_compression(
                 incoming_samples, update_tensor_meta=update_tensor_meta
             )
         return self.extend_if_has_space_image_compression(
             incoming_samples, update_tensor_meta=update_tensor_meta
         )
 
+    def extend_if_has_space_byte_compression_text(
+        self,
+        incoming_samples: np.ndarray,
+        update_tensor_meta: bool = True,
+        lengths: Optional[List[int]] = None,
+    ):
+        if lengths is None:
+            lengths = np.zeros(len(incoming_samples), dtype=np.uint32)
+            for i, sample in enumerate(incoming_samples):
+                lengths[i] = sample.__len__()  # assume ~1 bytes per char
+        sample_nbytes = np.mean(lengths)
```

Comment:
Chunk compression is not super accurate with chunk sizes anyway, so this is fair.